### PR TITLE
[Codegen] Make Python Class Naming More Robust

### PR DESCRIPTION
**Pull request by noanflaherty**
This handles some edge cases I came across while trying to pull down some workflows.

### CODE REVIEW

File: ee/codegen/src/__test__/nodes/__snapshots__/api-node.test.ts.snap [hunk 1]
```diff
@@ -2,13 +2,13 @@
 
 exports[`ApiNode > basic > getNodeDisplayFile 1`] = `
 "from vellum_ee.workflows.display.nodes import BaseAPINodeDisplay
-from ...nodes.api_node import ApiNode
+from ...nodes.api_node import APINode
```
Comment (noanflaherty) [inline ee/codegen/src/__test__/nodes/__snapshots__/api-node.test.ts.snap:5]: Pretty cool that it maintains consecutive caps-letters now.

File: ee/codegen/src/__test__/utils/cases.test.ts [hunk 2]
```diff
@@ -17,18 +21,35 @@ describe("Casing utility functions", () => {
...
   });
 
-  describe("toPascalCase", () => {
-    const testCases = [
-      { input: "hello-world", expected: "HelloWorld" },
-      { input: "hello_world", expected: "HelloWorld" },
-      { input: "hello world", expected: "HelloWorld" },
-      { input: "hello-world_example 123", expected: "HelloWorldExample123" },
-    ];
+  describe("createPythonClassName", () => {
```
Comment (noanflaherty) [inline ee/codegen/src/__test__/utils/cases.test.ts:24]: Thanks for the test cases cursor!

File: ee/codegen/src/utils/casing.ts [hunk 3]
```diff
@@ -7,10 +7,60 @@ export function toKebabCase(str: string): string {
     .toLowerCase(); // Convert to lowercase
 }
 
-export function toPascalCase(str: string): string {
-  return str
-    .split(/[-_\s]+/)
-    .map((word) => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
+export function createPythonClassName(input: string): string {
```
Comment (noanflaherty) [inline ee/codegen/src/utils/casing.ts:10]: Thanks even more for the gnarly implementation cursor! Lot's of talking back and forth to get this.

Comment (dvargasfuertes) [reply ee/codegen/src/utils/casing.ts:10]: you can tell when cursor writes a PR by the number of comments between each line

**Review by dvargas92495 — APPROVED** [review]

File: ee/codegen/src/utils/casing.ts (hunk 3 above)
Comment (kirawoods) [reply ee/codegen/src/utils/casing.ts:10]: So true

**Review by kirawoods — COMMENTED** [review]
✅
